library/game_state.py

```diff
@@ -131,15 +140,49 @@ def confirm_entry(self, term: blessed.Terminal) -> None:
 
         self.redraw_user_term(term)
 
-    def end_of_turn(self, term: blessed.Terminal) -> None:
+    def end_of_turn(self, term: blessed.Terminal, board: Board) -> None:
         """End of Turn"""
         self.current = 40
 
-        # update board
-        self.update_board = True
+        # Handle end of turn
+        working_space_location = convert_to_space_location(self.user_select_space)
+        subgrid_number = str(self.user_select_subgrid)
+        subgrid = board.collect_subgrid(subgrid_number)
+        if self.player_active == 1:
+            subgrid[working_space_location[0], working_space_location[1]] = "X"
+        elif self.player_active == 2:
+            subgrid[working_space_location[0], working_space_location[1]] = "O"

```

Comment:
```suggestion
        subgrid[working_space_location[0], working_space_location[1]] = "X" if self.player_active == 1 else "O"
```

or

```suggestion
        subgrid[working_space_location[0], working_space_location[1]] = ["·", "X", "O"][self.player_active]
```

if you're feeling unreadable ;)

(though of course if you prefer the current code that's fine too.)